test/runtests.jl

```diff
@@ -41,6 +41,22 @@ using UUIDs: uuid4
             @test rows[2].depends_on == rows[1].id
             @test rows[3].depends_on == nothing
             @test rows[4].depends_on == rows[3].id
+
+            nested_dags = (
+                DemoGenerator(1) => DemoGenerator(1) => DemoGenerator(1),
+                DemoGenerator(1) => DemoGenerator(1) => [DemoGenerator(1)],
+                DemoGenerator(1) => [DemoGenerator(1) => DemoGenerator(1)],
+                DemoGenerator(1) => [DemoGenerator(1) => [DemoGenerator(1)]],
+                DemoGenerator(1) => (DemoGenerator(1) => DemoGenerator(1)),  # tuple
```

Comment:
```suggestion
                DemoGenerator(1) => (DemoGenerator(1) => DemoGenerator(1),),  # tuple
```

Reply:
ah yeah good spot - I can see now how to fix that